### PR TITLE
ref: upgrade pre-commit

### PR DESCRIPTION
**Pull request by joshuarli**
Primarily, this upgrades pytest to 6.2.x which they've dropped support for Python < 3.6 and have pyupgraded their entire codebase (adds f-string and other stuff that makes things a bit faster), and also upgrades our own pyupgrade and reruns it, yielding some more f-strings and nice changes.

### CODE REVIEW

**Comment by joshuarli** [comment]
Need to upgrade attrs pin in getsentry. https://github.com/getsentry/getsentry/runs/2671647165?check_suite_focus=true

**Review by armenzg — APPROVED** [review]

File: src/sentry_plugins/asana/plugin.py [hunk 1]
```diff
@@ -143,7 +143,7 @@ def get_client(self, user):
     def error_message_from_json(self, data):
         errors = data.get("errors")
         if errors:
-            return " ".join([e["message"] for e in errors])
+            return " ".join(e["message"] for e in errors)
```
Comment (EvanPurkhiser) [inline src/sentry_plugins/asana/plugin.py:146]: Oh I like this

Comment (EvanPurkhiser) [reply src/sentry_plugins/asana/plugin.py:146]: It does this automatically now with black or something?

Comment (EvanPurkhiser) [reply src/sentry_plugins/asana/plugin.py:146]: Oh this is pyupgrade, nice

**Review by evanpurkhiser — APPROVED** [review]

File: src/sentry_plugins/asana/plugin.py (hunk 1 above)
Comment (joshuarli) [reply src/sentry_plugins/asana/plugin.py:146]: Yeah, it's good.

**Review by wedamija — APPROVED** [review]

**Comment by joshuarli** [comment]
Hmm, not able to reproduce these symbolicator test failures. They succeed locally.

**Comment by joshuarli** [comment]
Ok, just gonna back out of pytest upgrade for now.

**Pull request merged**